### ISSUE TITLE
Pulling Fuel from Central complains about missing Result, due to missing version

### ISSUE THREAD

**Issue by DanySK** [kittinunf/fuel#757]
# Bug Report

## Description

When using central and using Fuel this way:

```kotlin
repositories {
    mavenCentral()
}
dependencies {
    implementation("com.github.kittinunf.fuel:fuel:+")
    implementation("com.github.kittinunf.fuel:fuel-coroutines:+")
    implementation("com.github.kittinunf.fuel:fuel-gson:+")
}
```

Gradle complains:

```
* What went wrong:
Execution failed for task ':compileKotlin'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find com.github.kittinunf.result:result:2.2.0.
     Searched in the following locations:
       - https://repo.maven.apache.org/maven2/com/github/kittinunf/result/result/2.2.0/result-2.2.0.pom
     If the artifact you are trying to retrieve can be found in the repository but without metadata in 'Maven POM' format, you need to adjust the 'metadataSources { ... }' of the repository declaration.
     Required by:
         project : > com.github.kittinunf.fuel:fuel:2.2.3
```

This is caused by the fact that the first version of Result published on Central (thanks to kittinunf/result#70) is 3.0.1.

## Proposed solution

I suggest tackling this in two rather straightforwards steps:

1. Make the current Fuel depend on Result 3.0.1
2. Release a new version of Fuel

Since Central enforces a no-retract policy, it's impossible to change the current version to point to Result 3.0.1, a new release is thus necessary.

## Alternate solution

Manually releasing result 2.2.0 on Central would work as well, but that requires manual intervention, and thus precious human time.

## Temporary workaround

Who is currently using Fuel via Central can work around the issue by enforcing a compile-time dependency on Result (warning: it taints the dependency tree)
```kotlin
dependencies {
    implementation("com.github.kittinunf.result:result:3.0.1")
}
```

**Comment by DanySK** [comment]
Example work around here: https://github.com/kittinunf/fuel/issues/757

**Comment by kittinunf** [comment]
I will take care of this 🙇 sorry for inconveniences 🙏

**Comment by DanySK** [comment]
Thank you @kittinunf :pray: :bow:

**Comment by nfrankel** [comment]
Hello

Any way to help? I cannot override the dependency as a workaround, as I'm using Kotlin Scripting.

**Comment by kittinunf** [comment]
@nfrankel Let me help you out by manually pushing 2.2.0 to the maven central 🙇

**Comment by nfrankel** [comment]
Thanks a lot @kittinunf! However, in the meantime, I switched to OkHttp because I was blocked: the override workaround doesn't work in the context of Kotlin Scripting

**Comment by kittinunf** [comment]
@nfrankel Oh no, sorry for the delay :( I will try to be more promptly support next time 🙇

**Comment by nfrankel** [comment]
@kittinunf No worries! This is your personal Open Source project, I don't expect timely support.

Keep up the good work

**Comment by jayveeravalli** [comment]
@kittinunf any update on this? I am facing the same error for version 2.2.3

**Comment by kittinunf** [comment]
We have released new version 2.3.0 and it will reference the Result library (3.1.0) from the mavenCentral. It should take a few hours or so for everyone to be able to fetch! 🙇 🙇 🙇

**Comment by kittinunf** [comment]
I think this should solve all of the problem regarding to version on `mavenCentral`. 🙇 🙇 Feel free to re-open if this doesn't fix your issue. Otherwise, please let me close this. :)
